app/uk/gov/hmrc/traderservices/controllers/TraderServicesFrontendController.scala

```diff
@@ -89,6 +89,18 @@ class TraderServicesFrontendController @Inject() (
       whenAuthorisedWithForm(AsUser)(DeclarationDetailsForm)(Transitions.submittedDeclarationDetails)
     }
 
+  // GET /pre-clearance/export-questions
+  val showAnswerExportQuestions: Action[AnyContent] =
+    actionShowStateWhenAuthorised(AsUser) {
+      case _: AnswerExportQuestions =>
```

Comment:
Do we need a state here or is Unit ok? Is this just a case of forwarding on to the export-questions page?

Reply:
this partial function is a precondition on the controller side, if the user is in the right state the action will continue, otherwise, the user will be redirected to the right URL reflecting his current state

Reply:
if we put here `whenAuthorised(AsUser)(Transitions.showAnswerExportQuestions)(display)` then our business logic would prevent transition anyway but the former is specially designed to handle it